File: solutions/405.convert-a-number-to-hexadecimal/convert-a-number-to-hexadecimal.js
/**
 * @param {number} num
 * @return {string}
 */
var toHex = function(num) {
    const hexChars = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'a', 'b', 'c', 'd', 'e', 'f'];
    let res = '';
    res = hexChars[num & 15];
    num = num >>> 4;
    while(num !== 0){
        res = hexChars[num & 15] + res;
        num = num >>> 4;
    }
    return res;
};

File: solutions/405.convert-a-number-to-hexadecimal/convert-a-number-to-hexadecimal.java
class Solution {
    public String toHex(int num) {
        char[] map = {'0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'a', 'b', 'c', 'd', 'e', 'f'};
        String res = "";
        if(num == 0) return "0";
        while(num != 0){
            int digit = num & 15;
            res = map[digit] + res;
            num >>>= 4;
        }
        
        return res;
    }
}